src/gtest/test_keys.cpp

```diff
@@ -164,3 +164,60 @@ TEST(Keys, EncodeAndDecodeUnified)
         }
     }
 }
+
+TEST(Keys, EncodeAndDecodeUnifiedFullViewingKeys)
+{
+    SelectParams(CBaseChainParams::MAIN);
+    KeyIO keyIO(Params());
+
+    UniValue ua_tests = read_json(MAKE_STRING(json_tests::unified_full_viewing_keys));
+
+    for (size_t idx = 0; idx < ua_tests.size(); idx++) {
+        UniValue test = ua_tests[idx];
+        std::string strTest = test.write();
+        if (test.size() < 1) // Allow for extra stuff (useful for comments)
+        {
+            FAIL() << "Bad test: " << strTest;
+            continue;
+        }
+        if (test.size() == 1) continue; // comment
+
+        libzcash::UnifiedFullViewingKeyBuilder builder;
+        // ["p2pkh_key_bytes, sapling_key_bytes, orchard_key_bytes, unknown_key_bytes, unified_key"]
+        // These were added to the UA in preference order by the Python test vectors.
+        if (!test[0].isNull()) {
+            auto data = ParseHex(test[0].get_str());
+            ASSERT_EQ(data.size(), 65);
+            CDataStream ss(data, SER_NETWORK, PROTOCOL_VERSION);
+            auto decoded = CChainablePubKey::Read(ss);
+            ASSERT_TRUE(builder.AddTransparentKey(decoded));
+        }
+        if (!test[1].isNull()) {
+            auto data = ParseHex(test[1].get_str());
+            ASSERT_EQ(data.size(), 128);
+            CDataStream ss(data, SER_NETWORK, PROTOCOL_VERSION);
+            auto key = libzcash::SaplingDiversifiableFullViewingKey::Read(ss);
+            ASSERT_TRUE(builder.AddSaplingKey(key));
+        }
+
+        // Orchard keys and unknown items are not yet supported; instead,
+        // we just test that we're able to parse the unified key string
+        // and that the constituent items match the elements; if no Sapling
+        // key is present then UFVK construction would fail because it might
+        // presume the UFVK to be transparent-only.
```

Comment:
If there isn't already a relevant one, open an issue to add support for unknown components. Ideally we'd be testing right now with the Orchard components as unknown, since `UnifiedFullViewingKey` is intended to be generic.